src/Storages/Statistics/Statistics.h

```diff
@@ -83,6 +87,36 @@ class ColumnsDescription;
 using ColumnStatisticsPtr = std::shared_ptr<ColumnStatistics>;
 using ColumnsStatistics = std::vector<ColumnStatisticsPtr>;
 
+/**
+ * Statistics represent the statistics for a part, table or node in query plan.
+ * It is a computable data structure which means:
+ *   1. it can accumulate statistics for a dedicated column of multiple parts.
+ *   2. it can support more calculations which is needed in estimating complex predicates like `a > 1 and a < 100 or a in (1, 3, 500).
+ *      For example column 'a' has a minmax statistics with range [1, 100], after estimate 'a>50' the range is [51, 100].
+ */
+class Statistics
+{
+public:
+    explicit Statistics() : total_row_count(0) { }
+    explicit Statistics(Float64 total_row_count_) : total_row_count(total_row_count_) { }
+    explicit Statistics(Float64 total_row_count_, const std::unordered_map<String, ColumnStatisticsPtr> & column_stats_)
+        : total_row_count(total_row_count_), column_stats(column_stats_)
+    {
+    }
+
+    void merge(const Statistics & other);
+    void addColumnStat(const ColumnStatisticsPtr & column_stat_);
+
+    Float64 getTotalRowCount() const { return total_row_count; }
+    ColumnStatisticsPtr getColumnStat(const String & column_name) const;
+
+private:
+    Float64 total_row_count;
+    std::unordered_map<String, ColumnStatisticsPtr> column_stats;
```

Comment:
Please add a comment what is mapped to what? Otherwise, reader wonders what `String` stands for.

Reply:
And if we rename `ColumnStatistics` to `ColumnPartStatistics` (see my comment above), then we could rather name this `column_part_statistics`.